### DISCUSSION TITLE
Show height markers on invisible paths when doing path stuff

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#19599] [Feature requests]
Currently the only way to interact with invisible paths is to enable height markers on paths. It would be useful if height markers automatically appeared on paths when performing actions that build or remove paths: when placing a bench/lamp/bin, when building paths, when bulldozing a path, and when placing a peep/staff.

A quick poll of NEDesigns asking them if they wanted height marks to appear when placing path additions or when building paths:
- 12 voted for height marks appearing at both times
- 1 voted against the feature
- 0 voted for appearing in one but not the other

Implementation:
- a flag would be added to path FootpathSurfaceEntry to indicate height marks should appear when building (i.e. it's invisible)
- when one of the specified modes is triggered (placing a path addition, when the path build window is open, when placing a peep, when the bulldoze window is open and paths is selected) height markers will be drawn on paths whose FootpathSurfaceEntry require it.
- Tickboxes would be added next to the virtual floor dropdown in the options menu to control when they are displayed.

**Comment by cheweytoo** [comment]
That poll is kind of asking the wrong question 😉.

Since the "height markers allow interaction with invisible paths" thing is just kind of coincidental, it would make for _much_ cleaner UI/UX to make the actual paths visible (and thereby clickable) if path building or path addition building is active.

Also see https://github.com/OpenRCT2/OpenRCT2/issues/18547#issuecomment-1460128286 .

**Reply by spacek531** [reply]
> make the actual paths visible (and thereby clickable)

They are already visible. The game draws a sprite for them. That sprite just happens to have no pixels that you can click on.

**Reply by cheweytoo** [reply]
> > make the actual paths visible (and thereby clickable)
> 
> They are already visible. The game draws a sprite for them. That sprite just happens to have no pixels that you can click on.

Well, that makes them not visible ;-) – drawing something with pixels (like a building ghost e.g.) would make them not just logically but also visibly… uh… visible.

**Reply by spacek531** [reply]
The issue is what sprite to use if not the ones in the path itself. Some users don't have any paths selected except the invisible path, so it's not possible to rely on other objects.

**Reply by ocalhoun6** [reply]
Something similar to the virtual floor feature might be a good solution to this problem, maybe? Path tiles drawn in a *very* simple geometric style, drawn the same way the virtual floor is.

**Reply by Broxzier** [reply]
> The issue is what sprite to use if not the ones in the path itself. Some users don't have any paths selected except the invisible path, so it's not possible to rely on other objects.

There could be default path sprites that show up while placing paths. Something like a very simplistic, flat path without any details. No object has to be loaded for this.

**Reply by cheweytoo** [reply]
> There could be default path sprites that show up while placing paths. Something like a very simplistic, flat path without any details. No object has to be loaded for this.

That would be the definitely superior solution. While the "clickable height values" thing is a (very!) useful workaround right now, this would be pretty bad UX in general: Height values are not something that can be interacted with in any other case.

**Comment by ocalhoun6** [comment]
>A quick poll of NEDesigns asking them if they wanted height marks to appear when placing path additions or when building paths:

Just to note, though, the NE community might be a little biased on that choice and not necessarily be a good representation of the players as a whole. (Because the types of parks the NE guys do rely _heavily_ on invisible paths, but invisible paths are something that a lot of more casual players probably never mess with at all.)

And automatically adding height marks when those tools are open _does_ add quite a bit of visual clutter, which might be confusing or annoying to players who aren't using invisible paths at all. (And will probably result in some bug reports about the new behavior.)

I think ideally, it would only automatically turn the height marks on _if_ invisible paths are present in the park. (Or at least, if invisible paths are selected as an object that *could* be in the park.) That would ensure that it only happens when needed and doesn't clutter, confuse, or annoy players who aren't using invisible paths, while it would still be very convenient to those who are using invisible paths.

**Reply by spacek531** [reply]
It was worded poorly. They question was if they would appear *on invisible paths* when placing etc.

**Reply by ocalhoun6** [reply]
Ah, yeah. If the height markers *only* appear on invisible paths, that would be a great way to solve the problem.
